cloudera/datadog_checks/cloudera/api_client_v7.py

```diff
@@ -19,17 +20,38 @@ def collect_data(self):
         read_clusters_response = clusters_resource_api.read_clusters(cluster_type='any', view='full')
         self._log.debug("Cloudera full clusters response:\n%s", read_clusters_response)
 
+        if self._check.config.clusters:
+            self._log.debug("clusters config: %s", self._check.config.clusters)
+            d = Discovery(
+                lambda: clusters_resource_api.read_clusters(cluster_type='any', view='full').items,
+                limit=self._check.config.clusters.limit,
+                include=self._check.config.clusters.include,
+                exclude=self._check.config.clusters.exclude,
+                interval=self._check.config.clusters.interval,
+                key=lambda cluster: cluster.name,
+            )
+            discovered_clusters = list(d.get_items())
```

Comment:
Sorry if this is a stupid question, but there's on thing I don't understand here. Is the `collect_data` called each time an integration runs? In that case, shouldn't we store the `d` variable at the object level (in an attribute)? Because here we are going to recreate a new Discovery each and call the API each time and the cache won't be used

Reply:
`clusters_resource_api.read_clusters(cluster_type='any', view='full').items` can give us different results from a run to the next, and in that case `discovered_clusters` can be different

Reply:
So we do not want to use the cache?

Reply:
If it is configured, we will use it. But this is something internal to the `Discovery` class

Reply:
How is it used if we recreate the Discovery object each time the check runs?

Reply:
You are right. I misunderstood your first question. Changes done.